File: note-service/src/main/java/com/devflow/note/service/NoteService.java
package com.devflow.note.service;

import java.util.List;

import com.devflow.note.dto.CreateNoteRequest;
import com.devflow.note.dto.NoteResponse;
import com.devflow.note.dto.UpdateNoteRequest;

public interface NoteService {

    NoteResponse createNote(CreateNoteRequest request, String userEmail);

    List<NoteResponse> getAllNotes(String userEmail);

    NoteResponse getNoteById(Long id, String userEmail);

    NoteResponse updateNote(Long id, UpdateNoteRequest request, String userEmail);

    void deleteNote(Long id, String userEmail);
}

File: devflow-frontend/src/api/noteService.js
import axiosInstance from "./axiosInstance";

const getNotes = () => {
  return axiosInstance.get("/api/notes");
};

const createNote = (note) => {
  return axiosInstance.post("/api/notes", note);
};

const updateNote = (id, note) => {
  return axiosInstance.put(`/api/notes/${id}`, note);
};

const deleteNote = (id) => {
  return axiosInstance.delete(`/api/notes/${id}`);
};

export default {
  getNotes,
  createNote,
  updateNote,
  deleteNote,
};
